### ISSUE TITLE
warning: no visible global function definition in continuous integration

### ISSUE THREAD

**Issue by ElsLommelen** [jimhester/lintr#466]
Today I got a lot of warnings after running lintr during continuous integration on Wercker, which are not related to the few changes I committed (and yesterday CI passed):
`Warning: no visible global function definition for '...'`

The warnings mention either a function of the package itself or an imported function, which I added under `imports` in the description file, and after `@importFrom` in the `Roxygen2` documentation (and these functions were added to the namespace file).

I tried running `lintr::lint.package()` locally in the project folder of the package in RStudio, but this did not give any warnings.

[Here](https://github.com/inbo/LSVI/commit/b06be933d386c1546b21663f907aa596245f64a6) is the commit and [here](https://app.wercker.com/ElsLommelen/LSVI/runs/build/5e6789287b63df001aeadeaf?step=5e6789a0a15bef0008cefb7f) the report with warnings on Wercker.  Any idea what could be wrong?

**Comment by russHyde** [comment]
Does it pass `R CMD check` locally?

**Comment by ElsLommelen** [comment]
Yes, it passes R CMD check.

**Comment by russHyde** [comment]
It looks like a server problem: 
```
Bad Gateway (HTTP 502). Failed to {
  "message": "Server Error"
}
```

If it is reproducible, then it might be an issue with https://github.com/jimhester/wercker-step-r-lint rather than lintr itself.

**Comment by ElsLommelen** [comment]
Honestly I have no idea how to check if this is a server problem.
The warnings showed up again after each new commit.  For now I decided to ignore the object_usage_linter temporary to be able to continue working.
(To test reproducibility, I started dublicating the repo, but it turned out to be much more work than I expected.  I will continu next week.)

**Comment by russHyde** [comment]
Sorry, I think the server problem in your wercker run was unrelated to the issue you've raised.
I've just cloned LSVI and ran lintr on it, and have received the object-usage lints.

**Comment by ElsLommelen** [comment]
I started a new repo in which I already simplified the package a bit: [test_lintr_on_wercker](https://github.com/ElsLommelen/test_lintr_on_wercker).  I have not yet pinpointed the problem, but I already learnt that it is probably OS specific to Unix systems:
- On a windows computer no warning is given locally in `R CMD check` or `lintr::lint_package()` ([commits](https://github.com/ElsLommelen/test_lintr_on_wercker/commits/master) of 13 march)
- On a Linux computer, on the other hand, warnings may also show up locally in `R CMD check` or `lintr::lint_package()`.  This was the case in [this commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/0528c107d16417a8e8bfc6763e4a87e8ebe009fa), but due to solving other problems, the local check did not give the specific warnings before doing [this commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/d061d198f70b219588683f97de54e55a06e32158) (but it failed on wercker in a rocker/verse container anyway).

**Comment by ElsLommelen** [comment]
I simplified the repo [test_lintr_on_wercker](https://github.com/ElsLommelen/test_lintr_on_wercker) till I could pinpoint the problem: if a SQLite database and/or a connection to a SQLite database is present in the package, the warning in lintr shows up on a Unix OS for all functions preceded by a line of comments in the previous line.

The [one but last commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/06195e64617fbc9de83009705077d61f3a506f88) seems the most simplified case:
- the function `selecteerIndicatoren` has a line of comments and 2 imported functions that evoke a warning
- the package moreover contains a SQLite database and a function `connecteerMetLSVIdb` that makes a connection with this database (not used in ´selecteerIndicatoren´)

After removing the SQLite database and related functions in [the last commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/2b10d9f46fa65651a16115283668db56fa174e65), the warnings disappeared.

Previous commits show that the warning is given only when the comment is on the line preceding the funtion:
- in [this commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/17beffc9af8bbf5a971bec639b17febaf9802871) I added a comment on line 23 (and simplified the function) and I noticed the function on line 24 gave a warning, which was not the case in the [previous commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/fd7bf28c3afd9cd1c130c4cf18381aab8375e742)
- in [this commit](https://github.com/ElsLommelen/test_lintr_on_wercker/commit/fa246a35259ddfa5ab7e03c2beb92e0515ef6d70) a new warning shows up after moving `dbGetQuery()` from line 50 (2 lines after comment) to line 49 (line after comment)

There is one important question that I could not solve and that might lead to a solution: why did the warnings suddenly appear on the 10th of mars?  In the package itself there were no important changes that could explain their appearance.  That day no new version of the RSQLite package was published.

(In the package I switched to English for relevant information and comments on this issue.  So just ignore the Dutch parts; for my own convenience I did not change the function names but I think the above information is sufficient to understand it?)

I hope this helps to solve the problem?  Admittedly I still do not know in which package this issue should be solved.

**Comment by ElsLommelen** [comment]
In [this package](https://github.com/inbo/dhcurve) (which I hadn't looked for for years before I did some changes yesterday) I encounter the same problem (which first occurred in [PR 28](https://github.com/inbo/dhcurve/pull/28)), but in a slightly different setting:
- the main package with functions and documentation does not contain a database, but the inst-folder contains a script `hoofdscript.R` to retain data from an Access database (which is put there as an example for a limited group of users)
- here we use package `RODBC` to connect to the database, whereas we used `DBI` and `odbc` in the previously reported package

Admittedly it is an unusual situation to have a script with an Access database tested in a Linux environment, but I hope it can help to tackle the problem.  For now I will ignore the object_usage_linter for my own convenience.

**Comment by AshesITR** [comment]
Can you check if this was fixed by #740?
Additionally, I found it advisable to `devtools::load_all()` the to-be-linted package as well before running `lint_package()` - this ensures all functions and objects from all source files are visible to `checkUsage()`.

**Comment by ElsLommelen** [comment]
I checked both of my previously reported cases, and the warnings did not show up anymore.
Thanks for fixing this!

**Comment by AshesITR** [comment]
That's good to hear! Thanks for checking.
